administrative-sdk/controllers/choiceChallengeController.js

```diff
@@ -0,0 +1,112 @@
+const ChoiceChallenge = require('../models/choiceChallenge').ChoiceChallenge;
+
+/**
+ * Controller class for the ChoiceChallenge model.
+ */
+class ChoiceChallengeController {
+  /**
+   * @param connection Object to connect to.
+   */
+  constructor(connection) {
+    this.connection = connection;
+  }
+
+  /**
+   * Create a choice challenge.
+   *
+   * @param {ChoiceChallenge} choiceChallenge Object to create.
+   * @returns Promise containing the newly created object.
+   * @rejects If the server returned an error.
+   */
+  createChoiceChallenge(choiceChallenge) {
+    // Validate required domain model fields.
+    if (!choiceChallenge.organisationId) {
+      return Promise.reject(new Error('organisationId field is required'));
+    }
+    const url = this.connection.settings.apiUrl + '/organisations/' +
+      choiceChallenge.organisationId + '/challenges/choice';
+    const fd = new FormData();
+    if (choiceChallenge.id !== undefined &&
+      choiceChallenge.id !== null) {
+      fd.append('id', choiceChallenge.id);
+    }
+    fd.append('question', choiceChallenge.question);
+    choiceChallenge.choices.forEach(choice => {
+      fd.append('choices', choice);
+    });
+    return this.connection._secureAjaxPost(url, fd)
+      .then(data => {
+        // Update the id in case domain model didn't contain one.
+        choiceChallenge.id = data.id;
+        choiceChallenge.created = new Date(data.created);
+        choiceChallenge.updated = new Date(data.updated);
+        choiceChallenge.status = data.status;
+        choiceChallenge.choices = [];
+        data.choices.forEach(pair => {
+          choiceChallenge.choices.push(pair.choice);
+        });
+        return choiceChallenge;
+      });
+  }
+
+  /**
+   * Get a choice challenge.
+   *
+   * @param {Connection} connection Object to connect to.
+   * @param {string} organisationId Specify an organisation identifier.
+   * @param {string} challengeId Specify a choice challenge identifier.
+   * @returns Promise containing a ChoiceChallenge.
+   * @rejects If no result could not be found.
+   */
+  static getChoiceChallenge(connection, organisationId, challengeId) {
```

Comment:
Indeed I'm doing this in an upcoming PR. This changes functionality though so I kept it out of this PR.